Python/specialize.c

```diff
@@ -1957,6 +2008,93 @@ _Py_Specialize_CompareOp(PyObject *lhs, PyObject *rhs, _Py_CODEUNIT *instr,
     cache->counter = initial_counter_value();
 }
 
+char INIT_CLEANUP_CODE[8] = {
+    LOAD_ASSERTION_ERROR, 0,
+    RAISE_VARARGS, 1,
+    EXIT_INIT_CHECK, 0,
+    RETURN_VALUE, 0
+};
+
+PyFunctionObject *_Py_InitCleanupFunc = NULL;
+
+static int
+setup_init_cleanup_func(void) {
+    if (_Py_InitCleanupFunc != NULL) {
+        return 0;
+    }
+    PyObject *name = NULL;
+    PyObject *code = NULL;
+    PyObject *lines = NULL;
+    PyCodeObject *codeobj = NULL;
+    PyObject *globals = NULL;
+
+    name = _PyUnicode_FromASCII("type.__call__", strlen("type.__call__"));
+    if (name == NULL) {
+        goto cleanup;
+    }
+    code = PyBytes_FromStringAndSize(INIT_CLEANUP_CODE, 8);
+    if (code == NULL) {
```

Comment:
```suggestion
    if (code == NULL) {
        Py_DECREF(name);
```
It would leaks `name` otherwise.